flytepropeller/pkg/compiler/errors/compiler_errors.go

```diff
@@ -202,13 +202,17 @@ func NewDuplicateIDFoundErr(nodeID string) *CompileError {
 	)
 }
 
-func NewMismatchingTypesErr(nodeID, fromVar, fromType, toType string) *CompileError {
-	return newError(
-		MismatchingTypes,
-		fmt.Sprintf("Variable [%v] (type [%v]) doesn't match expected type [%v].", fromVar, fromType,
-			toType),
-		nodeID,
-	)
+func NewMismatchingTypesErr(nodeID, fromVar, fromType, toVar, toType string) *CompileError {
+	var errMsg string
+	if len(toVar) == 0 {
+		errMsg = fmt.Sprintf("Variable [%v] (type [%v]) doesn't match expected type [%v].", fromVar, fromType,
+			toType)
+	} else {
+		errMsg = fmt.Sprintf("The output variable '%v' has type [%v], but it's assigned to"+
+			" the input variable '%v' which has type type [%v].", fromVar, fromType, toVar, toType)
```

Comment:
Should we have 2 different functions instead? One to handle the previous case and another one to handle the new case where `toVar` is set.

Reply:
that's better. updated it, thanks~